src/pymovements/gaze/experiment.py

```diff
@@ -109,12 +120,27 @@ def __init__(
             origin=origin,
         )
 
-        checks.check_is_not_none(sampling_rate=sampling_rate)
-        assert sampling_rate is not None
+        checks.check_is_mutual_exclusive(sampling_rate=sampling_rate, eyetracker=eyetracker)
+
+        self.eyetracker = eyetracker
+
+        self._sampling_rate = sampling_rate
 
-        checks.check_is_greater_than_zero(sampling_rate=sampling_rate)
+        checks.check_is_not_none(sampling_rate=self.sampling_rate)
+        assert self.sampling_rate is not None
 
-        self.sampling_rate = sampling_rate
+        checks.check_is_greater_than_zero(sampling_rate=self.sampling_rate)
+
+    @property
+    def sampling_rate(self) -> float:
+        """Assign sampling rate to experiment."""
+        if self._sampling_rate is not None:
+            return self._sampling_rate
+
+        if self.eyetracker is not None:
+            return self.eyetracker.sampling_rate
+
+        raise TypeError("'self._sampling_rate' and 'self.eyetracker' must not both be None")
```

Comment:
I think this line can not be reached, as we already check in the constructor that the sampling rate is not None.

So you can just remove the raise and the last if-condition, and just `return self.eyetracker.sampling_rate`

Reply:
After removing the raise, mypy gave me the following error:
src/pymovements/gaze/experiment.py:140: error: Item "None" of "EyeTracker | None" has no attribute "sampling_rate"  [union-attr]
So, I added `# type: ignore [union-attr]` in line 140. Is this a viable approach?

Reply:
change the signature to `def sampling_rate(self) -> float | None:`